### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by tkoolen**
Includes the changes in https://github.com/mposa/drake/pull/20.

* merge zerostep and 1step into nstep
* improve comments in nstep
* add argument `n` to `LIPM_3D_nstep`
* recurse if necessary: calling `LIPM_3D_nstep(n)` will check for the existence of the .mat file for `n-1`-step, and if it doesn't exist, calls `LIPM_3D_nstep(n-1)`
* fix backoff for n-step

### CODE REVIEW

File: drake/examples/RobustBalancing/LIPM_3D_nstep.m [hunk 1]
```diff
@@ -1,112 +1,146 @@
...
+  [prog, goal_sos] = spotless_add_sprocedure(prog, V, V0p,V_vars,degree-2);
 
-% add goal region and h_X constraints
-[prog, sos(1)] = spotless_add_sprocedure(prog, sos(1), V0p,[W_vars;u],2);
-[prog, sos(1)] = spotless_add_sprocedure(prog, sos(1), h_X,[W_vars;u],degree);
-[prog, sos(2)] = spotless_add_sprocedure(prog, sos(2), h_X,V_vars,degree-2);
-[prog, sos(3)] = spotless_add_sprocedure(prog, sos(3), h_X,W_vars,degree-2);
-[prog, sos(4)] = spotless_add_sprocedure(prog, sos(4), h_X,W_vars,degree-2);
+  if time_varying
+    [prog, goal_sos] = spotless_add_sprocedure(prog, goal_sos, T^2-t^2,V_vars,degree-2);
```
Comment (tkoolen) [inline drake/examples/RobustBalancing/LIPM_3D_nstep.m]: I was wondering: why is this here? The step time shouldn't matter for 0-step capturability, right?

File: drake/examples/RobustBalancing/LIPM_3D_zerostep.m [hunk 2]
```diff
@@ -1,137 +0,0 @@
...
-[prog, sos(1)] = spotless_add_sprocedure(prog, sos(1), h_goal,V_vars,degree-2);
-[prog, sos(2)] = spotless_add_sprocedure(prog, sos(2), h_X,V_vars,degree-2);
-[prog, sos(3)] = spotless_add_sprocedure(prog, sos(3), h_X,W_vars,degree-2);
-[prog, sos(4)] = spotless_add_sprocedure(prog, sos(4), h_X,W_vars,degree-2);
-
-% 0 <= t < = 1
-% could also write this with two constraints
-if time_varying
-  [prog, sos(1)] = spotless_add_sprocedure(prog, sos(1), T^2-t^2,V_vars,degree-2);
-  [prog, sos(2)] = spotless_add_sprocedure(prog, sos(2), T^2-t^2,V_vars,degree-2);
```
Comment (tkoolen) [inline drake/examples/RobustBalancing/LIPM_3D_zerostep.m]: See comment above. This is where I got it from.

File: drake/examples/RobustBalancing/LIPM_3D_nstep.m [hunk 3]
```diff
@@ -1,112 +1,146 @@
...
-[prog, sos(1)] = spotless_add_sprocedure(prog, sos(1), step_max^2-u'*u,[W_vars;u],degree);
 
-% 0 <= t < = 1
+% (2) -Vdot(t,x) <= 0 for x in X
+[prog, Vdot_sos] = spotless_add_sprocedure(prog, -Vdot, h_X,V_vars,degree-2);
+% 0 <= t < = T
 % could also write this with two constraints
 if time_varying
-  [prog, sos(2)] = spotless_add_sprocedure(prog, sos(2), T^2-t^2,V_vars,degree-2);
+  [prog, Vdot_sos] = spotless_add_sprocedure(prog, Vdot_sos, T^2-t^2,V_vars,degree-2);
```
Comment (tkoolen) [inline drake/examples/RobustBalancing/LIPM_3D_nstep.m]: same here (if n == 0).

File: drake/examples/RobustBalancing/LIPM_3D_nstep.m (hunk 1 above)
Comment (tkoolen) [inline drake/examples/RobustBalancing/LIPM_3D_nstep.m]: I guess T just has a different meaning in zerostep?

Comment (tkoolen) [inline drake/examples/RobustBalancing/LIPM_3D_nstep.m]: Never mind, I got it.

Comment (mposa) [inline drake/examples/RobustBalancing/LIPM_3D_nstep.m]: This constraint (T^2 - t^2>=0) shouldn't apply to goal_sos, only to Vdot  (what used to be sos(2).
